File: server/php/src/Application/Domain/Payments/PaymentDetails/CustomerOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class CustomerOutput
{
    public ?CustomerDeviceOutput $device = null;

    public function toArray(): array
    {
        return [
            'device' => $this->device?->toArray(),
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/customer_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class CustomerOutput
          attr_accessor :device

          def initialize(device: nil)
            @device = device
          end

          def to_h
            {
              'device' => @device&.to_h
            }
          end
        end
      end
    end
  end
end